### DISCUSSION TITLE
Issues running pytest suite

### DISCUSSION THREAD

**Discussion by sniafas** [mlflow/mlflow#5741] [Q&A] [answered]
I am facing problems running pytest suite. 

1) running a pytests from shell script
```
$ ./dev/run-small-python-tests.sh

tests/__init__.py:1: in <module>
    from mlflow.utils.logging_utils import _configure_mlflow_loggers
mlflow/__init__.py:31: in <module>
    from mlflow.utils.logging_utils import _configure_mlflow_loggers
E     File "/home/projects/mlflow/mlflow/utils/logging_utils.py", line 90
E       print(*args, file=MLFLOW_LOGGING_STREAM, **kwargs)  # pylint: disable=print-function
E             ^
E   SyntaxError: invalid syntax
```
2) running pytest regular 

many tests are failing because of
`ModuleNotFoundError: No module named 'databricks_cli'`

Is there any workaround for this?

**Comment by harupy** [comment]
1. Run `python --version` to check the python version.
2. Run `pip install databricks_cli`

**Reply by sniafas** [reply]
Thank you @harupy 

In mlflow conda environment
1) Python 3.7.11
2) Requirement is already satisfied `(mlflow-dev-env)$ databricks -v: Version 0.16.7.dev0`

Still the problem persists

**Comment by harupy** [comment, marked as answer]
What does `which pytest` print out?

**Reply by sniafas** [reply]
There seems to be a mess
- (mlflow-dev-env) $ `which pytest` /usr/bin/pytest
- (base) $ `which pytest` home/user/anaconda3/bin/pytest

Using pytest from (base) solves 1), but 2) is still there because of being out of mlflow-dev-env 
:/ 

god i hate anaconda...

**Reply by harupy** [reply]
Did you install pytest in the mlflow-dev-env environment?

**Reply by sniafas** [reply]
Managed to purge everything and rebuild from scratch. Thanks for indicating where to look. Cheers!
